engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -2491,7 +2492,69 @@ namespace dmGameObject
             }
 
             if (!DispatchMessages(collection, &collection->m_ComponentSocket, 1))
+            {
                 ret = false;
+            }
+        }
+
+        if (update_context->m_FixedUpdateFrequency != 0)
+        {
+            const float time = collection->m_FixedAccumTime + update_context->m_DT; // Add the scaled time
+            const float fixed_frequency = update_context->m_FixedUpdateFrequency;
+            // If the proxy is slowed down, we want e.g. the physics to be slowed down as well
+            const float fixed_dt = (1.0f / (float)fixed_frequency) * update_context->m_TimeScale;
+            float num_fixed_steps = (uint32_t)(time / fixed_dt);
+            // Store the remainder for the next frame
+            collection->m_FixedAccumTime = time - (num_fixed_steps * fixed_dt);
+
+            if (num_fixed_steps != 0)
+            {
+                UpdateContext fixed_update_context;
+                fixed_update_context = *update_context;
+                fixed_update_context.m_DT = fixed_dt;
+
+                for (uint32_t step = 0; step < num_fixed_steps; ++step)
+                {
+                    for (uint32_t i = 0; i < component_types; ++i)
```

Comment:
Basically the same update loop as the regular update, just with a different timestep and function